bundle/Command/EzFlowMigrateCommand.php

```diff
@@ -111,33 +133,37 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $legacyPages = $legacyModel->getPages();
 
             Report::write('Reading legacy block configuration');
-            $blockMapper = new BlockMapper($legacyWrapper->getBlockConfiguration(), $this->getContainer()->get('twig'));
+            $blockMapper = new BlockMapper($legacyWrapper->getBlockConfiguration(), $this->twig);
 
             $configuration = [
                 'layouts' => [],
                 'blocks' => [],
                 'services' => [],
             ];
 
-            foreach ($legacyPages as $legacyPage) {
-                Report::write("Migrating page...");
-                Report::write("ContentId: {$legacyPage['contentobject_id']}, FieldId: {$legacyPage['id']}, Version: {$legacyPage['version']}");
+            $this->handler->beginTransaction();
+
+            try {
+                foreach ($legacyPages as $legacyPage) {
+                    Report::write('Migrating page...');
+                    Report::write("ContentId: {$legacyPage['contentobject_id']}, FieldId: {$legacyPage['id']}, Version: {$legacyPage['version']}");
 
-                $legacyPage['name'] = 'Migrated Landing Page';
-                $page = new Page($legacyPage, $this->getContainer(), $blockMapper);
+                    $legacyPage['name'] = 'Migrated Landing Page';
+                    $page = new Page($legacyPage, $this->xmlConverter, $this->blockTypeRegistry, $blockMapper);
 
-                $landingPage = $page->getLandingPage($configuration);
+                    $landingPage = $page->getLandingPage($configuration);
 
-                if ($landingPage) {
-                    Report::write("Save page as Landing Page");
+                    Report::write('Save page as Landing Page');
                     $legacyModel->updateEzPage($legacyPage['id'], $landingPage);
                 }
-                else {
-                    Report::write("Not valid page field (empty or used as call-to-action). Ignoring...");
-                }
+
+                $legacyModel->replacePageFieldType();
+            } catch (\Exception $e) {
+                $this->handler->rollBack();
+                throw $e;
             }
 
-            $legacyModel->replacePageFieldType();
+            $this->handler->commit();
```

Comment:
Usually `commit` is called as a last instruction inside `try` block.

Reply:
Thanks, good catch! Fixed in: https://github.com/ezsystems/ezflow-migration-toolkit/pull/5/commits/365be7830ee4b305d030bfee26bf27453750e030.